src/NCTablePad.cc

```diff
@@ -30,6 +30,53 @@
 #include <limits.h>
 
 
+void
+NCTableSortDefault::sort ( std::vector<NCTableLine *>::iterator itemsBegin,
+			   std::vector<NCTableLine *>::iterator itemsEnd )
+{
+    std::sort ( itemsBegin, itemsEnd, Compare(getColumn(), isReverse()) );
+}
+
+
+bool
+NCTableSortDefault::Compare::operator() ( const NCTableLine * first,
+					  const NCTableLine * second ) const
+{
+    std::wstring w1 = smartSortKey( first );
+    std::wstring w2 = smartSortKey( second );
+    wchar_t * endptr1 = 0;
+    wchar_t * endptr2 = 0;
+
+    long long number1 = std::wcstoll( w1.c_str(), &endptr1, 10 );
+    long long number2 = std::wcstoll( w2.c_str(), &endptr2, 10 );
+
+    if ( *endptr1 == L'\0' && *endptr2 == L'\0' )
+    {
+	// both are numbers
+	return !reverse ? number1 < number2 : number1 > number2;
```

Comment:
just NP, but usually it is more readable to be positive like `return reverse ? number1 > number2 : number1 < number2;`

Reply:
Well, since the function implements ```operator<``` I think its fine to have that comparison first.